apps/system/js/notifications.js

```diff
@@ -403,64 +404,80 @@ var NotificationScreen = {
+    if (!isNotificationAvailable) {
+      // We need to animate the ambient indicator when the toast
+      // timesout, so we skip updating it here, by passing a skip bool
+      this.addUnreadNotification(manifestURL, true);
+      var notificationNode = document.createElement('div');
+      notificationNode.classList.add('notification');
+      notificationNode.setAttribute('role', 'link');
+      notificationNode.setAttribute('id', manifestURL);
+      notificationNode.setAttribute('tabindex', '0');
+      notificationNode.dataset.notificationId = detail.id;
+      notificationNode.dataset.noClear = behavior.noclear ? 'true' : 'false';
+      notificationNode.lang = detail.lang;
+      notificationNode.dataset.predefinedDir = detail.bidi;
+      notificationNode.dataset.obsoleteAPI = 'false';
+      if (typeof detail.id === 'string' &&
+          detail.id.indexOf('app-notif-') === 0) {
+        notificationNode.dataset.obsoleteAPI = 'true';
+      }
+      var type = detail.type || 'desktop-notification';
+      notificationNode.dataset.type = type;
+      notificationNode.dataset.manifestURL = manifestURL;
+      // in case detail.icon value is empty, appIcon value is used.
+      if (detail.icon || detail.appIcon) {
+        var icon = document.createElement('img');
+        icon.src = detail.icon ? detail.icon : detail.appIcon ;
+        icon.setAttribute('role', 'presentation');
+        notificationNode.appendChild(icon);
+      }
+      var titleContainer = document.createElement('div');
+      titleContainer.classList.add('title-container');
+      var countSpan = document.createElement('span');
+      countSpan.classList.add('detail-count');
+      countSpan.textContent = '';
+      titleContainer.appendChild(countSpan);
+      var title = document.createElement('div');
+      title.classList.add('title');
+      this.getNotificationTitle(detail, title);
+      title.setAttribute('dir', 'auto');
+      titleContainer.appendChild(title);
+
+      var time = document.createElement('span');
+      var timestamp = detail.timestamp ? new Date(detail.timestamp) : new Date();
+      time.classList.add('timestamp');
+      time.dataset.timestamp = timestamp;
+      time.textContent = this.prettyDate(timestamp);
+      titleContainer.appendChild(time);
+      notificationNode.appendChild(titleContainer);
+
+      var message = document.createElement('div');
+      message.classList.add('detail');
+      var messageContent = document.createElement('div');
+      messageContent.classList.add('detail-content');
+      messageContent.textContent = this.getNotificationMessage(detail,'');
+      messageContent.setAttribute('dir', 'auto');
+      message.appendChild(messageContent);
+      notificationNode.appendChild(message);
+    } else {
+      // in case if notification is already present on the screen, its counter
+      // value will be incremented.
+      var notificationNode = document.querySelector('.notification');
```

Comment:
as debugging shows - it's not safe way to retrieve notification node - it will return first observed item with 'notification' in it's classList, and it easily won't be an item you created in ``if()`` branch above. So line 474 will cause error: ``JavaScript Error: "TypeError: notificationNode.querySelector(...) is null" `` and the remaining code in this function won't be processed.